File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
import java.util.*;

class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        List<Integer> list = new ArrayList<>();
        
        for(int i=1; i<=number; i++) {
            int count = 0;
            for(int j=1; j*j<=i; j++) {
                if(j*j==i) count++;
                else if(i%j==0) count+=2;
            }
            
            if(count>limit) count = power;
            
            list.add(count);
        } 
        
        for(int num : list) answer += num;
        
        return answer;
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.js
function solution(number, limit, power) {
    let answer = 0;
    for(let i = 1; i <= number; i++) {
        let count = 0;
        for(let j = 1; j * j <= i; j++) {
            if(i % j === 0) {
                count++;
                if(i / j != j) count++;
            }
        }
        if(count > limit) count = power;
        answer += count;
    }
    return answer;
}